scripts/debug_tools/reduce_repro_files.py

```diff
@@ -0,0 +1,107 @@
+import argparse
+import re
+import os
+import shutil
+import stat
+import subprocess
+import multiprocessing
+
+
+def get_file_path(analyzer_command_file):
+    with open(analyzer_command_file, 'r') as f:
+        return f.read().split(" ")[-1]
+
+
+def get_preprocessed_repro_file(abs_file_path, analyzer_command_file):
+    with open(analyzer_command_file, 'r') as f:
+        cmd = f.read().split(" ")
+        param_pattern = re.compile("-I|-D")
+        prepoc_params = [x for x in cmd if param_pattern.match(x)]
+        preproc = \
+            subprocess.Popen(["gcc", "-E"] + prepoc_params + [abs_file_path],
+                             stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        out, _ = preproc.communicate()
+        filename = abs_file_path.split('/')[-1]
+        # assuming exactly one dot (.) in the file name
+        prepoc_name = filename.split('.')[0] + "_preproc." + \
+            filename.split('.')[1]
+    with open(prepoc_name, 'w') as preproc_file:
+        preproc_file.write(out)
```

Comment:
I wonder if we could directly instruct the compiler to create the preprocessed file instead of writing this out to the standard output. This might be more efficient.